src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          if (state[property]) {
+            delete state[property];
+          }
+        }
+        break;
+
+      default:
+        const objKeys = Object.keys(state);
+
+        objKeys.forEach(key => {
+          delete state[key];
+        });
```

Comment:
It should not be a 'default' case
Simple loop will do it in one iteration, in your case its two